### PR TITLE
Add etherpad application

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

We need to deploy [etherpad](https://etherpad.org) instances that can be further integrated in our services.

## Proposal

- [x] add base application
- [x] test application in the CI

### CODE REVIEW

File: apps/etherpad/templates/services/app/dc.yml.j2
Comment (lunika) [inline apps/etherpad/templates/services/app/dc.yml.j2:54]: You are just testing is the application is up and running ? Should we do that in the probes without knowing why the application is failing  ?

File: apps/etherpad/templates/services/app/dc.yml.j2
Comment (lunika) [inline apps/etherpad/templates/services/app/dc.yml.j2]: should we use a variable for this path ? We are using it in different files, using a variable will prevent errors if this path changes

File: apps/etherpad/templates/services/app/dc.yml.j2
Comment (jmaupetit) [reply apps/etherpad/templates/services/app/dc.yml.j2:54]: I have no better idea at the time without modifying the application itself. Maybe we should consider integrating [terminus](https://github.com/godaddy/terminus) or [lightship](https://github.com/gajus/lightship) to add health check views to the express server but I am not sure if this can be achieved _via_ plugins development or just server script hacking.

Comment (jmaupetit) [reply apps/etherpad/templates/services/app/dc.yml.j2:54]: It looks like there is a server hook for that: https://etherpad.org/doc/v1.6.2/#index_expresscreateserver but we must create the plugin that integrates lightship.

Comment (jmaupetit) [reply apps/etherpad/templates/services/app/dc.yml.j2:54]: As this require some work to add a plugin to the image, if you agree, I would like to fix this in a separate PR. WDYT?

File: apps/etherpad/vars/vault/main.yml.j2
Comment (sampaccoud) [inline apps/etherpad/vars/vault/main.yml.j2:5]: why not separate the variables to make it simpler?

File: apps/etherpad/templates/services/app/dc.yml.j2
Comment (sampaccoud) [reply apps/etherpad/templates/services/app/dc.yml.j2:54]: So open an issue?

Comment (jmaupetit) [reply apps/etherpad/templates/services/app/dc.yml.j2:54]: I was waiting for @lunika answer :sweat_smile: Done, see #489

File: apps/etherpad/vars/vault/main.yml.j2
Comment (jmaupetit) [reply apps/etherpad/vars/vault/main.yml.j2:5]: If I you mean separating `user` from `password`, by doing so, you will duplicate the same request with the same complexity. Note that this request is used for almost every app vault file.

File: apps/etherpad/templates/services/app/dc.yml.j2
Comment (madmatah) [inline apps/etherpad/templates/services/app/dc.yml.j2]: You should add double quotes around these parameters to avoid problems with directories containing spaces.

File: apps/etherpad/templates/services/app/secret_keys.yml.j2
Comment (madmatah) [inline apps/etherpad/templates/services/app/secret_keys.yml.j2]: Should we fail instead of defining a default session key ?

File: apps/etherpad/templates/services/nginx/configs/healthcheck.conf.j2
Comment (madmatah) [inline apps/etherpad/templates/services/nginx/configs/healthcheck.conf.j2:14]: You should enable the nginx status page (see #479 )

File: apps/etherpad/templates/services/app/dc.yml.j2
Comment (jmaupetit) [reply apps/etherpad/templates/services/app/dc.yml.j2]: Please, don't tell me people are doing this?

Comment (madmatah) [reply apps/etherpad/templates/services/app/dc.yml.j2]: It's not forbidden !

File: apps/etherpad/templates/services/app/secret_keys.yml.j2
Comment (jmaupetit) [reply apps/etherpad/templates/services/app/secret_keys.yml.j2]: @sampaccoud asked me a similar question. IMO this should be a global policy for all vault-stored values.

Comment (jmaupetit) [reply apps/etherpad/templates/services/app/secret_keys.yml.j2]: But I can change this here if you insist a bit :sweat_smile:

Comment (madmatah) [reply apps/etherpad/templates/services/app/secret_keys.yml.j2]: If you change this here, it would show us the way to a better future :wink:

File: apps/etherpad/templates/services/nginx/configs/healthcheck.conf.j2
Comment (jmaupetit) [reply apps/etherpad/templates/services/nginx/configs/healthcheck.conf.j2:14]: Good catch!

File: apps/etherpad/templates/services/app/dc.yml.j2
Comment (jmaupetit) [reply apps/etherpad/templates/services/app/dc.yml.j2]: Done.

File: apps/etherpad/templates/services/app/secret_keys.yml.j2
Comment (jmaupetit) [reply apps/etherpad/templates/services/app/secret_keys.yml.j2]: Well said dude!

**Comment by jmaupetit** [comment]
@madmatah I've added fixup commits so that it's easier to give it a second review :pray:

**Review by madmatah — APPROVED** [review]

**Pull request merged**
